plugin/interestingwords.vim

```diff
@@ -197,29 +197,27 @@ function! s:buildColors()
   if (s:hasBuiltColors)
     return
   endif
-  if has('gui_running')
-    let ui = 'gui'
-    let wordColors = g:interestingWordsGUIColors
-  else
-    let ui = 'cterm'
-    let wordColors = g:interestingWordsTermColors
-  endif
+  let nGUIColors = len(g:interestingWordsGUIColors)
+  let nTermColors = len(g:interestingWordsTermColors)
+  let colorIndices = range(max([nGUIColors, nTermColors]))
```

Comment:
The repeated color may confuse the user who focus on the color rather than the word. For example, I highlight a word `uri` first, and then highlight word `url` at third. These two words look similar and both in same color. I may have trouble recognizing them.

Although the user could adjust their configuration to avoid problem. I think it is best to solve the problem in source codes.

That monstrosity is messy but useful. I suggest you wrap it into a function and refactoring it in future.